File: Cpp/Validate_Parentheses.cpp
#include<iostream>
#include<string.h>
using namespace std;

char stack[20];
int t=-1,n=20;

void push(int val){
  if(t>=n-1)
  cout<<"Stack Overflow!\n";
  else{
    t++;
    stack[t] = val;
  }
 }

void pop(){
  if(t<=-1) cout<<"Stack empty.\n";
  else  t--;
 } 


bool check(char *s){
  int size = strlen(s);
  char x;
  bool val=true;
  for(int i=0;i<size;i++){
    if(s[i]=='{' || s[i]=='[' || s[i]=='('){
      push(s[i]);
      continue;
    }
    else{
      switch(s[i]){
        case ')':
        {
          x = stack[t];
          pop();
          if(x=='[' || x=='{')
          val = false;
          break;
        }
        case ']':
        {
          x = stack[t];
          pop();
          if(x=='(' || x=='{')
          val = false;
          break;
        } 
        case '}':
        {
          x = stack[t];
          pop();
          if(x=='[' || x=='(')
          val = false;
          break;
        } 
      }
    }
  }
  if(t!=-1)
  val = false;
  return val;
 }

int main(){
  char test[20];
  cout<<"Enter sequence of parenthesis: ";
  cin>>test;
  if(check(test))
  cout<<"Balanced.\n";
  else
  cout<<"Not balanced.\n\n";  
  return 0;
 } 

File: Python/Validate_Parentheses.py
def validate(string):
    opening=['(','{','[']
    closing=[')','}',']']
    
    matching={'(':')','[':']','{':'}'}
    stack=[]
    flag=True
    count=0
    for s in string:
        #if bracket is opening
        if s in opening:
            stack.append(s)
        elif s in closing:
            #get topmost element
            try:
                top=stack.pop()
            except:
                #If the stack is empty
                flag=False
                break
            if matching[top]!=s:
                flag=False
                break
        else:
            flag=False
            break
    if len(stack)==0 and flag:
        return True
    else:
        return False

string=input("Enter String to check:")
if validate(string):
    print("Parentheses Matched")
else:
    print("Parentheses Not Matched")
        